src/CMakeLists.txt

```diff
@@ -143,6 +143,17 @@ if(APPLE AND (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_
         list(REMOVE_DUPLICATES CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
 endif()
 
+# Check for base threading library
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+if(${CMAKE_THREAD_LIBS_INIT} MATCHES "-pthread")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+elseif(CMAKE_THREAD_LIBS_INIT)
+  list(APPEND LIBS ${CMAKE_THREAD_LIBS_INIT})
+endif()
+
```

Comment:
Basically the only argument *against* using CMake's FindThread is that there is no way to tell it to *only* look for pthreads, but that is fixable, see rawspeed link